### PR TITLE
Remove project owners access group

### PR DESCRIPTION
**Pull request by dvoet**
prep for https://broadinstitute.atlassian.net/browse/GAWB-1245

a) stop using an extra (consistent but unnecessary) group for project owner access to workspaces, use project owner group directly
b) populate project owner intersection groups on workspace creation
c) change admin delete group api to specify the group name in the Url, not the payload.
d) other minor tweaks because I care

see also https://github.com/broadinstitute/firecloud-develop/pull/228

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell your tech lead (TL) that the PR exists if they want to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20161128_remove_project_owner_access_groups.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    <changeSet logicalFilePath="dummmy" author="dvoet" id="create_project_owner_groups_to_remove">
+        <createTable tableName="project_owner_groups_to_remove">
+            <column name="name" type="varchar(254)"></column>
+            <column name="email" type="varchar(254)"></column>
+        </createTable>
+    </changeSet>
+    <changeSet logicalFilePath="dummmy" author="dvoet" id="insert_project_owner_groups_to_remove">
+        <sql>
+            insert into project_owner_groups_to_remove select g.name, g.email from WORKSPACE_ACCESS wa inner join `GROUP` g on wa.group_name = g.name inner join GROUP_SUBGROUPS sg on sg.parent_name = g.name where wa.access_level='PROJECT_OWNER' and wa.is_realm_acl = 0;
```
Comment (breilly2) [inline core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20161128_remove_project_owner_access_groups.xml:12]: Is the join on `GROUP_SUBGROUPS` doing anything? I don't see any use of `sg` in the query and removing the join doesn't seem to affect (at least the number of rows in) the result in dev.

Comment (dvoet) [reply core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20161128_remove_project_owner_access_groups.xml:12]: once upon a time maybe I used a column from there...

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 2]
```diff
@@ -280,9 +277,9 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
           case wf if !WorkflowStatuses.withName(wf.status).isDone => dataAccess.workflowQuery.updateStatus(wf, WorkflowStatuses.Aborted)
         }}
 
-        //Gather the Google groups to remove
-        groupRefs: Set[RawlsGroupRef] = workspaceContext.workspace.accessLevels.values.toSet ++ workspaceContext.workspace.realmACLs.values
-        groupsToRemove <- DBIO.sequence(groupRefs.toSeq.map(groupRef => dataAccess.rawlsGroupQuery.load(groupRef)))
+        //Gather the Google groups to remove, but don't remove project owners group which is used by other workspaces
```
Comment (breilly2) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:282]: Should the project owners group be deleted in the case where the last workspace that references it is deleted? Or is there some other type of entity referencing it?

Comment (dvoet) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala:282]: the billing project owns the group, so no

File: core/src/test/scala/org/broadinstitute/dsde/rawls/integrationtest/HttpGoogleServicesDAOSpec.scala [hunk 3]
```diff
@@ -228,11 +226,7 @@ class HttpGoogleServicesDAOSpec extends FlatSpec with Matchers with IntegrationT
     //check that the owner is a part of both the access group and the intersection group
     Await.result(retry(when500)(() => Future { directory.members.get(ownerAG.groupEmail.value, gcsDAO.toProxyFromUser(testCreator)).execute() }), Duration.Inf)
     Await.result(retry(when500)(() => Future { directory.members.get(ownerIG.groupEmail.value, gcsDAO.toProxyFromUser(testCreator)).execute() }), Duration.Inf)
-
-    val projectOwnerAGMembers = Await.result(retry(when500)(() => Future { directory.members().list(projectOwnerAG.groupEmail.value).execute() }), Duration.Inf)
-    projectOwnerAGMembers.getMembers.map(_.getEmail) should be { Seq(projectOwnerGoogleGroup.groupEmail.value) }
-    val projectOwnerIGMembers = Await.result(retry(when500)(() => Future { directory.members().list(projectOwnerIG.groupEmail.value).execute() }), Duration.Inf)
-    projectOwnerIGMembers.getMembers.map(_.getEmail) should be { Seq(projectOwnerGoogleGroup.groupEmail.value) }
+    Await.result(retry(when500)(() => Future { directory.members.get(projectOwnerIG.groupEmail.value, gcsDAO.toProxyFromUser(testCreator)).execute() }), Duration.Inf)
```
Comment (breilly2) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/integrationtest/HttpGoogleServicesDAOSpec.scala:232]: In this case, you kept the Google call but removed the `should`. Was that intentional or is there something that could be checked about the result?

**Comment by breilly2** [comment]
:+1: Thanks for being patient while I learn enough about the system to feel like my thumbs-up has some value.

**Pull request merged**